### PR TITLE
feat: added rule for MySQL Scientific Notation bypass & update `sql-errors.data`

### PR DESCRIPTION
**Pull request by Xhoenix**
Added detection rule for MySQL Scienctific Notation Payloads
Reference: 

1) https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/SQL%20Injection/MySQL%20Injection.md#scientific-notation
2) https://blog.h3xstream.com/2021/10/bypassing-modsecurity-waf.html
3) #2318

### CODE REVIEW

**Comment by dune73** [comment]
Thanks for this contribution @Xhoenix.

You use a number that points to an existing rule family (-> we do this for stricter siblings).
Please use a non-used rule block. Suggestion: 942560.

What do you need the `capture` for?

The rule is only useful at PL1 (since we have existing rules catching this at PL2). But the combination of chars is so simple, I fear for FPs. Maybe we need to discuss this a bit.

**Comment by Xhoenix** [comment]
Didn't notice the `capture` :sweat_smile: 

Will update the rule number.

**Comment by Xhoenix** [comment]
@RedXanadu LGTM :)

**Comment by dune73** [comment]
There is still the `capture` action in your new rule. Otherwise looks good.

**Comment by Xhoenix** [comment]
Updated. :)

**Comment by dune73** [comment]
Than you very much.

Unfortunately it seems you might have to rebase / update the PR to catch up with recent merges. Otherwise this looks good and ready to be merged.

**Comment by Xhoenix** [comment]
Looks like the Mobile app is catching up, even code editing is now available. :) 

Couldn't find PR update option, though.

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf [hunk 1]
```diff
@@ -560,6 +560,35 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
...
 
+# This rule catches on Scientific Notation bypass payloads in MySQL
+# Reference: https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/SQL%20Injection/MySQL%20Injection.md#scientific-notation
+#
+# Regular expression generated from regex-assembly/942560.ra.
+# To update the regular expression run the following shell script
+# (consult https://coreruleset.org/docs/development/regex_assembly/ for details):
+#   crs-toolchain regex update 942560
+#
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?i)1\.e[\(-\),]" \
```
Comment (dune73) [inline rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:571]: @theseion Why do we get the `-` in this regex based on the source file above? Any why at this position?

**Comment by dune73** [comment]
This looks good to me. But I am a bit puzzled by the regex generated. 

@Xhoenix : You did not manipulate it by hand, did you?

**Comment by Xhoenix** [comment]
Yeah, I also found it weird and unnecessary, but as the toolchain added it I let it be.

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf (hunk 1 above)
Comment (theseion) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:571]: That's Go's regular expression library. They prefer to write ranges in character classes, which sometimes leads to this output. It's semantically correct.

**Review by theseion — APPROVED** [review]

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf (hunk 1 above)
Comment (dune73) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:571]: Fascinating. So this is a range, but the range only has two elements?

Comment (theseion) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:571]: Correct.

Comment (dune73) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:571]: Thanks

**Pull request merged**